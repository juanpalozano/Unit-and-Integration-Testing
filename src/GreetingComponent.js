// Funcional Components
//import React, {Fragment} from "react";
import React from "react";
import PropTypes from "prop-types";
const GreetingComponent = ({greetingTitle, greetingSubtitle}) => {
    return (
        <div>
            <h1>{greetingTitle}</h1>
            <p>{greetingSubtitle}</p>
        </div>
    );    
}
GreetingComponent.propTypes = {
    greetingTitle: PropTypes.string.isRequired
}
GreetingComponent.defaultProps = {
    greetingTitle: 'Default greeting title.',
    greetingSubtitle: 'Default greeting subtitle.'
}
export default GreetingComponent;