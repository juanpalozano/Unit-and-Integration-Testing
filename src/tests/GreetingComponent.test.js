import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GreetingComponent from '../GreetingComponent';
//import { render } from '@testing-library/react';


describe('Test File: GrettingComponent.js', () => {
    /*
    Jest Version
    test('Must show message: "Hello Developer!"', () =>{
        const greetingTitle = 'Hello Developer!';
        const { getByText } = render(<GreetingComponent greetingTitle={greetingTitle} />);
        expect(getByText(greetingTitle)).toBeInTheDocument();
    });
    test('Must show message: "多Are you ready to code today?"', () =>{
        const greetingSubtitle = '多Are you ready to code today?';
        const { getByText } = render(<GreetingComponent greetingSubtitle={greetingSubtitle} />);
        expect(getByText(greetingSubtitle)).toBeInTheDocument();
    });
    */
    test('The component should display the title "Hello Developer!".', () => {
        const greetingTitle = 'Hello Developer!';
        const wrapper = shallow(<GreetingComponent greetingTitle={greetingTitle} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('The component should display the subtitle "多Are you ready to code today?".', () => {
        const greetingSubtitle = '多Are you ready to code today?';
        const wrapper = shallow(<GreetingComponent greetingSubtitle={greetingSubtitle} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('The component should display the subtitle send from props.', () => {
        const greetingSubtitle = 'This is a test subtitle.';
        const wrapper = shallow(
            <GreetingComponent
                greetingSubtitle={greetingSubtitle}
            />
        );
        const subtitleText = wrapper.find('p').text();
        expect(subtitleText).toBe(greetingSubtitle);
    });
});