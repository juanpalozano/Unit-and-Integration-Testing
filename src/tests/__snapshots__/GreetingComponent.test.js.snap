// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test File: GrettingComponent.js The component should display the subtitle "¿Are you ready to code today?". 1`] = `
<div>
  <h1>
    Default greeting title.
  </h1>
  <p>
    ¿Are you ready to code today?
  </p>
</div>
`;

exports[`Test File: GrettingComponent.js The component should display the title "Hello Developer!". 1`] = `
<div>
  <h1>
    Hello Developer!
  </h1>
  <p>
    Default greeting subtitle.
  </p>
</div>
`;
